doc/api/errors.md

```diff
@@ -573,6 +580,79 @@ an argument of the wrong type has been passed to a Node.js API.
+An error with the `'ERR_INVALID_FILE_URL_HOST'` code may be thrown when a
+Node.js API that consumes `file:` URLs (such as certain functions in the
+[`fs`][] module) encounters a file URL with an incompatible host. Currently,
+this situation can only occur on Unix-like systems, where only `localhost` or
+an empty host is supported.
+
+<a id="ERR_INVALID_FILE_URL_PATH"></a>
+### ERR_INVALID_FILE_URL_PATH
+
+An error with the `'ERR_INVALID_FILE_URL_PATH'` code may be thrown when a
+Node.js API that consumes `file:` URLs (such as certain functions in the
+[`fs`][] module) encounters a file URL with an incompatible path. The exact
+semantics for determining whether a path can be used is platform-dependent.
+
+<a id="ERR_INVALID_THIS"></a>
+### ERR_INVALID_THIS
+
+The `'ERR_INVALID_THIS'` error code is used generically to identify that a
+Node.js API function is called with an incompatible `this` value. Its use is
+currently limited to the [WHATWG URL API][] to satisfy specification
+requirements, but may be expanded to other areas of the Node.js API in the
+future.
+
+Example:
+
+```js
+const { URLSearchParams } = require('url');
+const urlSearchParams = new URLSearchParams('foo=bar&baz=new');
+
+const buf = Buffer.alloc(1);
+urlSearchParams.has.call(buf, 'foo');
+  // Throws a TypeError with code 'ERR_INVALID_THIS'
+```
+
+<a id="ERR_INVALID_TUPLE"></a>
+### ERR_INVALID_TUPLE
+
+An error with code `'ERR_INVALID_TUPLE'` is thrown when an element in the
+`iterable` provided to the [WHATWG][WHATWG URL API] [`URLSearchParams`
+constructor][`new URLSearchParams(iterable)`] does not represent a name/value
+tuple -- that is, if an element is not iterable, or does not consist of exactly
+two elements.
+
+<a id="ERR_INVALID_URL"></a>
+### ERR_INVALID_URL
+
+An error using the `'ERR_INVALID_URL'` code is thrown when an invalid URL is
+passed to the [WHATWG][WHATWG URL API] [`URL` constructor][`new URL(input)`] to
+be parsed. The thrown error object typically has an additional property
+`'input'` that contains the URL that failed to parse.
+
+<a id="ERR_INVALID_URL_SCHEME"></a>
+### ERR_INVALID_URL_SCHEME
+
+The code `'ERR_INVALID_URL_SCHEME'` is used generically to signify an attempt
+to use a URL of an incompatible scheme (aka protocol) for a specific purpose.
+It is currently only used in the [WHATWG URL API][] support in the [`fs`][]
+module (which only accepts URLs with `'file'` scheme), but may be used in other
+Node.js APIs as well in the future.
+
+<a id="ERR_MISSING_ARGS"></a>
+### ERR_MISSING_ARGS
+
+The `'ERR_MISSING_ARGS'` error code is a generic error code for instances where
+a required argument of a Node.js API is not passed. This is currently only used
+in the [WHATWG URL API][] for strict compliance with the specification (which
+in some cases may accept `func(undefined)` but not `func()`). In most native
+Node.js APIs, `func(undefined)` and `func()` are treated identically, and the
+[`ERR_INVALID_ARG_TYPE`][] error code may be used instead.
```

Comment:
`'ERR_INVALID_ARG_TYPE'` has the following templated message:

> The "FOO" argument must be of type BAR.

In this case, it is not the question of what type the argument is, but whether it is defined at all.